docs/source/topics/provenance/caching.rst

```diff
@@ -82,6 +94,31 @@ Process nodes
 The hashing of *Process nodes* is fixed and can only be influenced indirectly via the hashes of their inputs.
 For implementation details of the hashing mechanism for process nodes, see :ref:`here <internal_architecture:engine:caching>`.
 
+
+.. _topics:provenance:caching:control-hashing:calcjobs-parsers:
+
+Calculation jobs and parsers
+............................
+
+.. versionadded:: 2.6
+    Resetting the calculation job cache
+
+    When the implementation of a ``CalcJob`` or ``Parser`` plugin changes significantly, it can be the case that for identical inputs, significantly different outputs are expected.
```

Comment:
Might be nice to add a few practical examples here to make sure the plugin developers understand when they should bump the cache version: e.g.

- Adding an output node to the parser
- Changing an existing output node (given identical input nodes, identical log files).
- Changing defaults in Calcjob inputs for a QM computation that are part of the QM input file, but not part of AiiDA input nodes (although I guess this should be discouraged in general?). I could imagine e.g. tightening WF convergence thresholds or something like that?